assets/environment-interpreter/activate/activate

```diff
@@ -269,13 +270,21 @@ esac
 # TODO: we could restore _start_env from the prior activation when performing an ephemeral activation
 
 # sets _FLOX_ATTACH, _FLOX_ACTIVATION_STATE_DIR, _FLOX_ACTIVATION_ID
-# Don't eval on one line so that we exit if flox-activations fails
-to_eval="$($_flox_activations \
-  start-or-attach \
-  --runtime-dir "$FLOX_RUNTIME_DIR" \
-  --pid "$$" --flox-env "$FLOX_ENV" --store-path "$_FLOX_ACTIVATE_STORE_PATH")"
-eval "$to_eval"
-export _FLOX_ACTIVATION_STATE_DIR _FLOX_ACTIVATION_ID
+# Only set if we are invoked by the flox CLI, which will only be the
+# case if _FLOX_WATCHDOG_BIN is set.
+if [ -n "${_FLOX_WATCHDOG_BIN:-}" ]; then
+  # Don't eval on one line so that we exit if flox-activations fails
+  to_eval="$($_flox_activations \
+    start-or-attach \
+    --runtime-dir "$FLOX_RUNTIME_DIR" \
+    --pid "$$" --flox-env "$FLOX_ENV" --store-path "$_FLOX_ACTIVATE_STORE_PATH")"
+  eval "$to_eval"
+  export _FLOX_ACTIVATION_STATE_DIR _FLOX_ACTIVATION_ID
+else
+  _FLOX_ATTACH="false"
+  _FLOX_ACTIVATION_STATE_DIR="$(mktemp -d)"
+  export _FLOX_ACTIVATION_STATE_DIR
+fi
```

Comment:
blocking: this makes it such that you can't attach if you `podman exec` to a running container

Reply:
So what solution do you propose? We can use a different condition to test when performing a build and select on that instead. I note that the presence of `$FLOX_SRC_DIR` could be used, but I'd think it's better to avoid env variables for this (and I've heard you say the same). We could add a new `--no-watchdog` flag or similar, but then again that's kind of the same as not providing an argument to the existing `--watchdog` flag (which is what this code is currently testing on).

Reply:
We could rename --turbo -> --build
Don't attach
Don't start a watchdog
No services
No user shell

Reply:
Thanks - that should work!  👍

Reply:
@mkenigs you repurposed a new `--mode (build|build-sandbox)` so I'll resolve this for now.